kernel/src/process/signal/poll.rs

```diff
@@ -158,17 +158,12 @@ impl Poller {
         }
     }
 
-    /// Wait until there are any interesting events happen since last `wait`. The `wait`
-    /// can be interrupted by signal.
-    pub fn wait(&self) -> Result<()> {
-        self.event_counter.read(&self.waiter, None)?;
-        Ok(())
-    }
-
-    /// Wait until there are any interesting events happen since last `wait` or a given timeout
-    /// is expired. This method can be interrupted by signal.
-    pub fn wait_timeout(&self, timeout: &Duration) -> Result<()> {
-        self.event_counter.read(&self.waiter, Some(timeout))?;
+    /// Waits until some interesting events happen since the last wait or until the timeout
+    /// expires.
+    ///
+    /// The waiting process can be interrupted by a signal.
+    pub fn wait(&self, timeout: Option<&Duration>) -> Result<()> {
+        self.event_counter.read(&self.waiter, timeout)?;
```

Comment:
We originally have both `wait` and `wait_timeout`, as a `timeout` is often not provided. This way, users are not required to pass `None` when a `timeout` isn't specified. Our `WaitQueue` follows a similar pattern, offering both `wait_until` and `wait_until_or_timeout`. This approach involves a trade-off: more APIs are required, but the call sites are simplified.

What are your thoughts on maintaining a consistent API style with the existing code? This PR also introduces `timeout` as a parameter for `Pollable`. To be consistent with the API style, `Pollable` should include both `wait_events` and `wait_events_or_timeout`.

Reply:
To be honest, I don't have much thoughts on this problem. If you have some preference, I can follow your instruction to address this problem.

Below are some feelings, but no conclusions:
 - There will be a lot of duplicate APIs if we do this: `EventCounter::read{,_timeout}`, `Poller::wait{,_timeout}`, `Pollable::wait_events{,_or_timeout}`.
 - It doesn't necessarily "simplify" the call sites: Many system calls (e.g. `poll()`, `send()`/`recv()` [via `SO_RCVTIMEO`/`SO_SNDTIMEO`](https://man7.org/linux/man-pages/man7/socket.7.html)) conditionally specify timeouts. In such scenarios, if our APIs don't support conditionally accepting the timeout, it will complicate the call sites.
 - Our waiting API primitives have two variants (with and without timeout), but some system calls optionally provide a timeout. The question is, where is the right place to unify the divergence? We can do it in the implementation of _each_ system call (before this PR), or in a common API (introduced in this PR, since it's a common fact of multiple system calls).

Reply:
On second thought, I think for `WaitQueue` we have to implement `wait` or `wait_or_timeout` separately because the former lives in OSTD but the latter lives in the Asterinas kernel.

However, for other APIs that live in the Asterinas kernel, I just realized that we might be able to do the following:
```rust
fn pause(&self) -> Result<()> {
    self.pause_or_timeout(self, Duration::MAX);
}

fn pause_or_timeout(&self, timeout: Duration) -> Result<()> {
    todo!()
}
```

This API interface seems a bit clearer? But of course it requires a separate PR and is not suitable for this PR.

Reply:
> This API interface seems a bit clearer? But it may require a separate PR and is not suitable for this PR.

Sure, such an API interface is clear for me. Of course, we can delay it to future PRs. 

> Our waiting API primitives have two variants (with and without timeout), but some system calls optionally provide a timeout.

Yes, if timeout is optional, then it seems to be reasonable to provide an API that can accept `Option<Duration>` as parameter.